execution/client/ethclient/engine.go

```diff
@@ -36,28 +36,22 @@ import (
 /*                                 NewPayload                                 */
 /* -------------------------------------------------------------------------- */
 
-// NewPayload calls the engine_newPayloadV3 method via JSON-RPC.
+// NewPayload calls the appropriate version of the Engine API NewPayload method.
 func (s *Client) NewPayload(
 	ctx context.Context,
 	payload *ctypes.ExecutionPayload,
 	versionedHashes []common.ExecutionHash,
 	parentBlockRoot *common.Root,
 ) (*engineprimitives.PayloadStatusV1, error) {
-	// This is wrong. We should do bytes comparison of versions
-	// Setting this to version.Deneb1 but we should fix version
-	// comparison and drop this check here. The EL does not know
-	// anything about CL versions, so payload should not have a
-	// CL version (or at least not a check here).
-	if payload.Version() == version.Deneb ||
-		payload.Version() == version.Deneb1 {
-		return s.NewPayloadV3(
-			ctx, payload, versionedHashes, parentBlockRoot,
-		)
+	// newPayloadV3 is used for beacon versions Deneb and onwards.
```

Comment:
I think @rezbera idea is to move the fork versions check out if this.
Rationale being that EL does not really know about versions so even ethclient engine should not?

I like the idea but it does not have to be implemented in this PR

Reply:
this is actually preserving the existing behavior so I'd rather keep the version enforcement in this PR. If we choose to remove it completely, can do in a separate PR